File: maximum-average-subarray-i/maximum_average_subarray_i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        if k == 1 or len(nums) == 1:
            return max(nums)/1.0
        maxKSum = currentSum = sum(nums[:k])
        for i in range(len(nums)-k):
            currentSum = currentSum - nums[i] + nums[i+k]
            maxKSum = max(currentSum, maxKSum)
        return maxKSum/k*1.0

File: maximum-average-subarray-i/maximum_average_subarray_i.go
func findMaxAverage(nums []int, k int) float64 {
	currentSum, maxKSum := nums[0], nums[0]
	if k == 1 {
		for _, n := range nums {
			if maxKSum < n {
				maxKSum = n
			}
		}
	} else {
		for _, n := range nums[1:k] {
			currentSum += n
		}
		maxKSum = currentSum
		for i := 0; i < len(nums)-k; i++ {
			currentSum = currentSum - nums[i] + nums[i+k]
			if maxKSum < currentSum {
				maxKSum = currentSum
			}
		}
	}
	return float64(maxKSum) / float64(k)
}
